lib/classes/batch/class-batch-task-manager.php

```diff
@@ -0,0 +1,347 @@
+<?php
+/**
+ * Batch task manager
+ *
+ * @since 4.0.0
+ */
+
+namespace wpCloud\StatelessMedia\Batch;
+
+// Require lib classes if not yet available
+if (!class_exists('UDX_WP_Async_Request')) {
+  require_once ud_get_stateless_media()->path('lib/ns-vendor/classes/deliciousbrains/wp-background-processing/classes/wp-async-request.php', 'dir');
+}
+
+if (!class_exists('UDX_WP_Background_Process')) {
+  require_once ud_get_stateless_media()->path('lib/ns-vendor/classes/deliciousbrains/wp-background-processing/classes/wp-background-process.php', 'dir');
+}
+
+use wpCloud\StatelessMedia\Helper;
+use wpCloud\StatelessMedia\Singleton;
+
+class BatchTaskManager extends \UDX_WP_Background_Process {
+  use Singleton;
+  
+  const STATE_KEY = '_state';
+  const UPDATED_KEY = '_updated';
+  const WAIT_FORCE_INTERVAL = 60 * 5; // 5 minute
```

Comment:
@balexey88 let's add new CONSTANT for that, if it is set - use it, if not - use default value

Reply:
Added constant [`WP_STATELESS_BATCH_HEALTHCHECK_INTERVAL`](https://github.com/udx/wp-stateless/wiki/Constants#wp_stateless_batch_healthcheck_interval)

The minimum value is 5 minutes, otherwise there is a risk that the batch will start simultaneously several times.